py_rql/filter.py

```diff
@@ -0,0 +1,93 @@
+#
+#  Copyright © 2022 Ingram Micro Inc. All rights reserved.
+#
+from cachetools import LFUCache
+
+from py_rql import parse
+from py_rql.cast import get_default_cast_func_for_type
+from py_rql.constants import FilterLookups, FilterTypes, RESERVED_FILTER_NAMES
+from py_rql.exceptions import (
+    RQLFilterLookupError,
+    RQLFilterNotFoundError,
+)
+from py_rql.transformer import RQLToFunctionTransformer
+
+
+class Filter:
```

Comment:
I don't like to discuss naming, but here I think that it's important. I had the same issue with Django-RQL.
I would suggest `FilterClass`. Please, ping me in Zoom for me to provide the reasoning.